octavia_f5/tests/unit/controller/worker/test_l2_sync_manager.py

```diff
@@ -86,32 +171,223 @@ def test_ensure_l2_flow_override_host(self, mock_get_network,
+
+    @mock.patch("octavia.network.drivers.noop_driver.driver.NoopManager"
+                ".get_subnet")
+    def test__do_ensure_l2_flow_nothing_exist_route_domain_failed_on_second_device(
+            self, mock_get_subnet):
+        mock_get_subnet.return_value = network_models.Subnet(
+            id='test-subnet-id', gateway_ip='1.2.3.1',
+            cidr='1.2.3.0/24', network_id='test-network-id')
+        mock_network = f5_network_models.Network(
+            mtu=8950, id='test-network-id', subnets=['test-subnet-id'],
+            segments=[{'provider:physical_network': 'physnet',
+                       'provider:segmentation_id': 1234}]
+        )
+        selfip_fixed_ip = network_models.FixedIP(
+            ip_address='1.2.3.2', subnet_id='test-subnet-id-2')
+        selfip_port = network_models.Port(
+            id='test-selfip-port-id', fixed_ips=[selfip_fixed_ip],
+        )
+        mock_bigip_1 = mock.Mock(spec=as3restclient.AS3RestClient)
+        mock_bigip_1.hostname = 'hostname-1'
+        mock_bigip_1.is_active = True
+        mock_bigip_1.get.side_effect = [
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 404),
+            MockResponse({}, 200),
+        ]
+        mock_bigip_2 = mock.Mock(spec=as3restclient.AS3RestClient)
+        mock_bigip_2.hostname = 'hostname-2'
+        mock_bigip_2.get.side_effect = [
+            MockResponse({}, 404) for _ in range(8)]
+        mock_bigip_2.post.side_effect = [
+            # VLAN creation
+            MockResponse({}, 202),
+            # Route Domain creation
+            MockResponse({}, 502)
+        ]
+        data = {
+            'hostname-1': {
+                'selfips': [selfip_port],
+                'store': {
+                    'network': mock_network,
+                    'bigip': mock_bigip_1,
+                    'subnet_id': 'test-subnet-id',
+                    'existing_selfips': []
+                }
+            },
+            'hostname-2': {
+                'selfips': [selfip_port],
+                'store': {
+                    'network': mock_network,
+                    'bigip': mock_bigip_2,
+                    'subnet_id': 'test-subnet-id',
+                    'existing_selfips': []
+                }
+            }
+        }
+        # self.manager._do_ensure_l2_flow(data=data)
+        self.assertRaises(Exception, self.manager._do_ensure_l2_flow, data=data)
+        # check thath both devices were called and REVERT tasks were also called
+        self.assertEqual(mock_bigip_1.get.call_count, 10)
+        self.assertEqual(mock_bigip_2.get.call_count, 7)
+        self.assertEqual(mock_bigip_1.post.call_count, 5)
```

Comment:
I don't know why, but this assertion fails with a call count of 4 instead of 5.

Reply:
I do not understand this comment. Do you want to see the requests list?

Reply:
Apparently this test is indeed flapping. I used a one-liner[1] to run this test 200 times and I got 13 failed runs. Five of them failed in line 371 with 8 instead of the expected 10 calls, the other eight failed in line 373 with 4 instead of 5 calls.

[1] `t=0;while true;do pytest octavia_f5/tests/unit/controller/worker/test_l2_sync_manager.py::TestL2SyncManager::test__do_ensure_l2_flow_nothing_exist_route_domain_failed_on_second_device |& grep '/test_l2_sync_manager.py", line 3\|MismatchError';t=$((t+1));echo $t;done`

Reply:
I copied your script and run it, waited for number `500` and I did not see any errors. I can record a video from my screen if you want. So, my guess this is your local environment failing. Did you run it inside docker container? Did you use correct python version in this container?

Reply:
Here it is:

https://github.com/user-attachments/assets/c795e7f9-bd0a-4ba8-ae92-635a08ae6f91

Reply:
Hmm, thanks for trying to reproduce this. This is weird. Apparently you're running this in a docker container? I'm running it in a local venv, so it might indeed be an issue with my setup, I will investigate. I'm using python 3.8.10 in that venv, just like in our base image.
For now I will treat this as an error on my side and not an issue with this PR 👍

Reply:
Yes, using exactly the same docker container with the real base container used in production - is the only way to check correctly that the code is working. I usually jump to the directory with git repo and run:
```
docker run -it --rm -v $(pwd):/usr/src/octavia-f5-provider-driver -w /usr/src/octavia-f5-provider-driver -e UPPER_CONSTRAINTS=https://raw.githubusercontent.com/sapcc/requirements/stable/yoga-m3/upper-constraints.txt -e VIRTUALENV_PIP="20.2.3" python:3.8 bash
```
then install Octavias and the plugin inside like it should be: https://github.com/sapcc/octavia-f5-provider-driver/blob/stable/yoga-m3/.github/workflows/pythonpackage.yml#L32-L33

and use this container as testing environment.